apigateway/phantom-token/access.lua

```diff
@@ -219,20 +189,22 @@ function _M.run(config)
         local access_token = string.gsub(access_token_untrimmed, "%s+", "")
         local result = verify_access_token(access_token, config)
     
+        -- Return technical error responses
         if result.status == 500 then
             error_response(ngx.HTTP_INTERNAL_SERVER_ERROR, 'server_error', 'Problem encountered authorizing the HTTP request')
         end
 
-        if result.status ~= 200 then
-            ngx.log(ngx.WARN, 'Received a ' .. result.status .. ' introspection response due to the access token being invalid or expired')
-            unauthorized_error_response(config)
+        -- Success responses forward a JWT to the target API
+        if result.status == 200 then
+            ngx.req.set_header('Authorization', 'Bearer ' .. result.jwt)
         end
 
-        ngx.req.set_header('Authorization', 'Bearer ' .. result.jwt)
-    else
-
-        ngx.log(ngx.WARN, 'No valid access token was found in the HTTP Authorization header')
-        unauthorized_error_response(config)
+        -- For invalid token errors, allow the target API to return a full 401 response with www-authenticate details
```

Comment:
I don't think this is a good approach. The API gateway already validates the access token via introspection and should not forward an invalid request to the API. Imo, it's the responsibility of the API gateway (not necessarily in this plugin) to return the header and resource metadata URL for the MCP endpoint.

Reply:
Still, APIs need to handle race conditions where tokens expire and return www-authenticate. I updated comments to explain that it could be done in the gateway and you have choices. We had a chat how at a team level, the www-authenticate logic could be the responsibility of developers and not gateway teams.